.github/workflows/contracts-test.yaml

```diff
@@ -0,0 +1,63 @@
+name: contracts-test
+on:
+  push:
+    paths:
+      - 'contracts/**'
+    branches:
+      - main
+  pull_request:
+    paths:
+      - 'contracts/**'
+    branches:
+      - main
+jobs:
+  hardhat-test:
+    runs-on: ubuntu-latest
+    container: node:lts
+    services:
+      sapphire-dev-ci:
+        image: ghcr.io/oasisprotocol/sapphire-dev:latest
+        ports:
+          - 8545
+          - 8546
+        env:
+          OASIS_DEPOSIT: /oasis-deposit -to 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
+        options: >-
+          --rm
+          --health-cmd="/oasis-node debug control wait-ready -a unix:/serverdir/node/net-runner/network/client-0/internal.sock"
+          --health-start-period=90s
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Install Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - uses: pnpm/action-setup@v2.2.4
+        name: Install pnpm
+        id: pnpm-install
+        with:
+          version: 7
+          run_install: false
+      - name: Get pnpm store directory
+        id: pnpm-cache
+        run: |
+          echo "::set-output name=pnpm_cache_dir::$(pnpm store path)"
+      - uses: actions/cache@v3
+        name: Setup pnpm cache
+        with:
+          path: ${{ steps.pnpm-cache.outputs.pnpm_cache_dir }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
+      - name: Install dependencies
+        run: pnpm --filter '@oasisprotocol/sapphire-*' install
+      - name: Build JS client
+        run: pnpm build
+        working-directory: clients/js
```

Comment:
I'd put `working-directory` above the `run` command. Same below.

Reply:
This has been modified